bioptim/examples/getting_started/example_soft_contact.py

```diff
@@ -0,0 +1,206 @@
+
+    # Problem parameters
+
+    tau_min, tau_max, tau_init = -100, 100, 0
+
+    # Add objective functions
+    objective_functions = ObjectiveList()
+    objective_functions.add(ObjectiveFcn.Lagrange.MINIMIZE_CONTROL, key="tau", weight=1)
+    objective_functions.add(ObjectiveFcn.Lagrange.MINIMIZE_SOFT_CONTACT_FORCES, weight=0.0001)
+    objective_functions.add(
+        ObjectiveFcn.Mayer.SUPERIMPOSE_MARKERS,
+        node=Node.START,
+        first_marker="marker_point",
+        second_marker="start",
+        weight=10,
+        axes=2,
+    )
+    objective_functions.add(
+        ObjectiveFcn.Mayer.SUPERIMPOSE_MARKERS,
+        node=Node.END,
+        first_marker="marker_point",
+        second_marker="end",
+        weight=10,
+    )
+
+    # Dynamics
+    dynamics = DynamicsList()
+    dynamics.add(DynamicsFcn.TORQUE_DRIVEN)
+
+    # Constraints
+    constraints = ConstraintList()
+    constraints.add(
+        ConstraintFcn.SUPERIMPOSE_MARKERS, node=Node.START, first_marker="marker_point", second_marker="start"
+    )
+    constraints.add(ConstraintFcn.SUPERIMPOSE_MARKERS, node=Node.END, first_marker="marker_point", second_marker="end")
+
+    # Path constraint
+    x_bounds = BoundsList()
+    x_bounds.add(bounds=QAndQDotBounds(biorbd_model))
+    nQ = biorbd_model.nbQ()
+    x_bounds[0].min[:nQ, 0] = X0[:nQ] - slack
+    x_bounds[0].max[:nQ, 0] = X0[:nQ] + slack
+    x_bounds[0].min[nQ:, 0] = -slack
+    x_bounds[0].max[nQ:, 0] = +slack
+
+    # Initial guess
+    x_init = InitialGuessList()
+    x_init.add(initial_guess=X0, interpolation=InterpolationType.CONSTANT)
+
+    # Define control path constraint
+    u_bounds = BoundsList()
+    u_bounds.add([tau_min] * biorbd_model.nbGeneralizedTorque(), [tau_max] * biorbd_model.nbGeneralizedTorque())
+
+    u_init = InitialGuessList()
+    u_init.add([tau_init] * biorbd_model.nbGeneralizedTorque())
+
+    return OptimalControlProgram(
+        biorbd_model,
+        dynamics,
+        n_shooting,
+        final_time,
+        x_init,
+        u_init,
+        x_bounds,
+        u_bounds,
+        objective_functions,
+        constraints,
+        ode_solver=ode_solver,
+        use_sx=use_sx,
+        n_threads=n_threads,
+    )
+
+
+def main():
+    """
+    Defines a multiphase ocp and animate the results
+    """
+    model = "models/soft_contact_sphere.bioMod"
+    ode_solver = OdeSolver.RK8()
+    tf = 1
+    ns = 100
+    ocp = prepare_single_shooting(model, ns, tf, ode_solver)
+
+    # Find equilibrium
+    X = InitialGuess([0, 0.10, 0, 1e-10, 1e-10, 1e-10])
+    U = InitialGuess([0, 0, 0])
+
+    sol_from_initial_guess = Solution(ocp, [X, U])
+    s = sol_from_initial_guess.integrate(shooting_type=Shooting.SINGLE, continuous=True)
+    # s.animate()
+
+    # Rolling Sphere at equilibrium
+    x0 = s.states["q"][:, -1]
+    dx0 = [0] * 3
+    X0 = np.concatenate([x0, np.array(dx0)])
+    X = InitialGuess(X0)
+    U = InitialGuess([0, 0, -10])
```

Comment:
Unless you are changing these values in the test, I'd put these in the prepare_ocp to reduce the number of parameter to call the ocp

Reply:
done.